presto-iceberg/src/test/java/com/facebook/presto/iceberg/TestIcebergFileWriter.java

```diff
@@ -0,0 +1,181 @@
+        }
+
+        @Override
+        public boolean canCoerce(Type actualType, Type expectedType)
+        {
+            throw new UnsupportedOperationException();
+        }
+
+        private List<Type> getTypes()
+        {
+            return ImmutableList.of(BooleanType.BOOLEAN, INTEGER, BIGINT, DoubleType.DOUBLE, VARCHAR, VARBINARY, TIMESTAMP, DATE, HYPER_LOG_LOG);
+        }
+    }
+
+    @BeforeClass
+    public void setup() throws Exception
+    {
+        ConnectorId connectorId = new ConnectorId("iceberg");
+        SessionPropertyManager sessionPropertyManager = new SessionPropertyManager();
+
+        sessionPropertyManager.addConnectorSessionProperties(
+                connectorId,
+                ImmutableList.of(
+                        dataSizeSessionProperty("parquet_writer_page_size", "Parquet: Writer page size", new DataSize(10, DataSize.Unit.KILOBYTE), false),
+                        dataSizeSessionProperty("parquet_writer_block_size", "Parquet: Writer block size", new DataSize(10, DataSize.Unit.KILOBYTE), false),
+                        new PropertyMetadata<>(
+                                "parquet_writer_version",
+                                "Parquet: Writer version",
+                                VARCHAR,
+                                ParquetProperties.WriterVersion.class,
+                                ParquetProperties.WriterVersion.PARQUET_2_0,
+                                false,
+                                value -> ParquetProperties.WriterVersion.valueOf(((String) value).toUpperCase()),
+                                ParquetProperties.WriterVersion::name),
+                        new PropertyMetadata<>(
+                                "compression_codec",
+                                "The compression codec to use when writing files",
+                                VARCHAR,
+                                HiveCompressionCodec.class,
+                                HiveCompressionCodec.NONE,
+                                false,
+                                value -> HiveCompressionCodec.valueOf(((String) value).toUpperCase()),
+                                HiveCompressionCodec::name)));
+
+        Session session = testSessionBuilder(sessionPropertyManager)
+                .setCatalog(ICEBERG_CATALOG)
+                .setSchema("tpch")
+                .build();
+
+        this.connectorSession = session.toConnectorSession(connectorId);
+        TypeManager typeManager = new TestingTypeManager();
+        this.hdfsContext = new HdfsContext(connectorSession);
+        HdfsEnvironment hdfsEnvironment = IcebergDistributedTestBase.getHdfsEnvironment();
+        this.icebergFileWriterFactory = new IcebergFileWriterFactory(hdfsEnvironment, typeManager,
+                new FileFormatDataSourceStats(), new NodeVersion("test"), new OrcFileWriterConfig(), HiveDwrfEncryptionProvider.NO_ENCRYPTION);
```

Comment:
Why do we have Orc config settings here when we're suing Parquet?

Reply:
The iceberg file writer factory requires orc settings to be instantiated